src/platform/entity.ts

```diff
@@ -39,14 +47,42 @@ export namespace Entity {
     properties: {
       version: { type: 'string', enum: ['v3'] },
       id: { type: 'string', oneOf: [IPFSv1.schema, IPFSv2.schema] },
-      type: { type: 'string' },
+      type: { type: 'string', enum: Object.values(EntityType) },
       pointers: { type: 'array', items: { type: 'string', minLength: 1 } },
       timestamp: { type: 'number', minimum: 0 },
       content: { type: 'array', items: ContentMapping.schema },
-      metadata: { type: 'object', nullable: true }
+      metadata: {
+        type: 'object',
+        nullable: true,
+        required: []
+      }
+    },
+    if: { properties: { type: { const: EntityType.PROFILE } } },
+    then: { properties: { metadata: Profile.schema } },
+    else: {
+      if: { properties: { type: { const: EntityType.SCENE } } },
+      then: { properties: { metadata: Scene.schema } },
+      else: {
+        if: { properties: { type: { const: EntityType.WEARABLE } } },
+        then: { properties: { metadata: Wearable.schema } },
+        else: {
+          if: { properties: { type: { const: EntityType.STORE } } },
+          then: { properties: { metadata: Store.schema } },
+          else: {
+            if: { properties: { type: { const: EntityType.EMOTE } } },
+            then: { properties: { metadata: Emote.schema } }
+          }
+        }
+      }
+    },
```

Comment:
Maybe something like this works here to improve readability:

```
{
  "allOf": [
    {
      "if": { ... },
      "then": { ... }
    },
    {
      "if": { ... },
      "then": { ... }
    }
  ]
}
```



Pointed out [here](https://stackoverflow.com/a/54712494/3716394)